File: core/java/src/main/java/hara/lang/protocol/ICount.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.icount", name = "ICount")
public interface ICount {
  @HaraMethod(value = "count", arity = 1)
  long count();
}

File: core/rust/src/lang/protocol/icount.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.icount", name = "ICount")]
pub trait ICount {
    #[hara_method(value = "count", arity = 1, whole_wasm)]
    fn count(&self) -> usize;
}
